submissions/shramkoweb/js-dom/main.js

```diff
@@ -0,0 +1,104 @@
+const frontendFrameworks = [
+    {
+        id: 0,
+        name: 'React',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg',
+    },
+    {
+        id: 1,
+        name: 'Vue',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/9/95/Vue.js_Logo_2.svg',
+    },
+    {
+        id: 2,
+        name: 'Angular',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/c/cf/Angular_full_color_logo.svg',
+    },
+    {
+        id: 3,
+        name: 'Django',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/7/75/Django_logo.svg',
+    },
+    {
+        id: 4,
+        name: 'Ruby',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/f/f1/Ruby_logo.png',
+    },
+    {
+        id: 5,
+        name: 'Flask',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/3/3c/Flask_logo.svg',
+    },
+    {
+        id: 6,
+        name: 'Express',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/6/64/Expressjs.png',
+    },
+    {
+        id: 7,
+        name: 'JQuery',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/8/81/JQuery_logo_text.svg',
+    },
+];
+
+const getItemByID = (items, id) => items.find(item => item.id === id);
+const clearHTML = node => node.innerHTML = '';
+
+const renderListItem = ({ name, id }, activeID) => {
+    const item = document.createElement('li');
+    const button = document.createElement('button');
+    button.innerText = name;
+    button.type = 'button';
+    button.id = id;
+    button.className = activeID === id ? 'active button' : 'button';
+    item.append(button);
+
+    return item;
+};
+
+const renderListItems = (items, activeID) => {
+    const frameworksListElement = document.querySelector('.frameworks-list');
+    clearHTML(frameworksListElement);
+    const fragment = document.createDocumentFragment();
+    items.forEach((item) => fragment.append(renderListItem(item, activeID)));
+
+    frameworksListElement.append(fragment);
+};
+
+const renderContent = (activePageID) => {
+    const { url } = getItemByID(frontendFrameworks, activePageID);
+    const contentElement = document.querySelector('.content');
+    const image = document.createElement('img');
+    image.src = url;
+
+    clearHTML(contentElement);
+    contentElement.append(image);
+};
+
+const render = (state) => {
```

Comment:
When I didn't think about the performance. thanks =)

Reply:
Maybe @MaxKalashnyk talks about that renderListItems is redundant in render function. You need render list of buttons only once, when init your app. Then you must rerender only content (image in your  case)

Reply:
I Agree with @babayK0 , i think you must call renderListItems only once (first render).

Reply:
@babayK0 @Aldegid in this task i use MVC model.
And when my state(model) change i need rerender my app whit new state(new activePageID).

I think in this case its ok. When we move renderListItems out each render func its will be not MVC

Reply:
@Shramkoweb, it's really cool that you tried the MVC pattern. I respect your decision and the code you wrote. But I guess it will be better to use it in more complex apps. I recommend looking at this course:
https://www.udemy.com/course/the-complete-javascript-course/?admitad_uid=6151191e959be7d1516e751c1c354aae
(maybe you know about it as well)

It seems that using the MVC pattern here is quite overengineering.
This task is pretty similar to the tabs UI element and I'm not sure that the easiest way of how to design it will be the MVC pattern.
Unfortunately, I didn't get any of your suggestions on how to simplify your state and rendering logic.

Let's simplify it to the toggling of the ID of the framework and just toggle the src of the image of the selected framework.

Reply:
U really think I need rewrite all my app, because rerender button list.
I think I correct make my app, and fix issues.
In MVC pattern normal to rerender some DOM on state change.
I agree it's overwhelming, but I try MVC and make it's correct.

Reply:
@Shramkoweb, it's up to you. This task doesn't require to build it using the MVC pattern. 
I'm here to help to simplify this code and make it more readable for other people.
Sorry, I can't approve this code only because you used the MVC. 
Thanks for your understanding.

Reply:
I will resolve this one, if you don't mind.

@MaxKalashnyk thank you for detailed explanation and link to the course. This is a great peace of knowledge which should be considered in the future.
@Shramkoweb nicely done with separating of concerns between rendering and state management!